rewrite-core/src/main/java/org/openrewrite/RecipeScheduler.java

```diff
@@ -92,6 +94,18 @@ public RecipeRun scheduleRun(Recipe recipe,
         );
     }
 
+    private boolean hasScanningRecipe(Recipe recipe) {
+        if (recipe instanceof ScanningRecipe) {
+            return true;
+        }
+        for (Recipe r : recipe.getRecipeList()) {
```

Comment:
The `getRecipeList()` method can now get called multiple times, but I guess that is unproblematic for this use case. Or can we iterate over a "resolved" recipe tree?

Reply:
That's just a listing of recipes, not an execution of them.